arrow/src/test/java/org/apache/calcite/adapter/arrow/ArrowAdapterTest.java

```diff
@@ -278,13 +273,12 @@ static void initializeArrowState(@TempDir Path sharedTempDir)
     if (Bug.CALCITE_6294_FIXED) {
       plan = "PLAN=ArrowToEnumerableConverter\n"
           + "  ArrowProject(intField=[$0], stringField=[$1])\n"
```

Comment:
how can both sides be changed?
The above bug should ensure that one side is never executed.

Reply:
@mihaibudiu I have implemented the in operator, but the implementation of the in operator will be based on the implementation of the or operator. Since these are two separate jira cases, I have to submit two separate PRs.

in addition，`SEARCH($0, Sarg[0, 1, 2]) `will be converted to `OR(=($0, 0), =($0, 1), =($0, 12))` in translateAnd.
In a way, this is not a bug.

Reply:
this is not my question.
My question is "why do both cases of the if statement need to change?"

Reply:
@mihaibudiu Sorry, I misunderstood.
Here is my explanation.

`where in (0, 1, 2)` will be translated into `ArrowFilter(condition=[SEARCH($0, Sarg[0, 1, 2])]) `in the filter, and then translated into `OR(=($0, 0), =($0, 1), =($0, 2))`.
The same goes for `where "intField"=12 or "intField"= 11`, which will be translated into `ArrowFilter(condition=[SEARCH($0, Sarg[11, 12])])`, and then translated into `OR(=($0, 11), =($0, 12))`

Since my changes support `OR(=($0, 0), =($0, 1), =($0, 2) `and similar OR operators, it seems that they are all effective.

Reply:
I added tests related to the or operator to verify